eng/WpfArcadeSdk/tools/GenApi.targets

```diff
@@ -4,13 +4,30 @@
                       Condition="'$(_GenerateReferenceAssemblySource)'=='true'"/>
   </ItemGroup>
 
-  <PropertyGroup Condition="'$(_GenerateReferenceAssemblySource)'=='true'">
+  <PropertyGroup Condition="'$(_GenerateReferenceAssemblySource)'=='true'
+                                or $(MSBuildProjectName.Contains('_wpftmp'))">
     <!-- 
       By default, GenAPI's target (GenerateReferenceAssemblySource) generates code into $(TargetDir).
       Instead, WPF generates into the ref directory in the project's source tree.
     -->
     <GenAPITargetDir>$(MSBuildProjectDir)ref\</GenAPITargetDir>
+
+    <!-- 
+      Sets GenAPITargetDir for _wpftmp assemblies, these directories are not needed
+      and removed after generation.
+      This is a quick hack to avoid failures during GenerateReferenceAssemblySource for 
+      _wpftmp assemblies. Added this because once we set GenerateReferenceAssemblySource
+      parameter from CLI, it is enabled for temp assemblies as well, and we fail to filter
```

Comment:
Did you try just calling the `GenerateReferenceAssemblySource` target?  That target will just run GenAPI from your already built assembly.  That's how folks use this in runtime.

Reply:
I will give it a try. I made these changes a year back but never merged it; I have forgotten some of the details by now. I will take a look and make necessary updates.